js/components/Leaderboard.js

```diff
@@ -1,17 +1,68 @@
 /**
  * Leaderboard Component
  * Displays ranked list of all users by total pages read
+ * Supports filtering by league
  */
 
-window.Leaderboard = ({ users, currentProfile }) => {
+window.Leaderboard = ({ 
+  users, 
+  currentProfile,
+  leagues,
+  currentLeagueId,
+  setCurrentLeagueId,
+  leagueLeaderboard
+}) => {
   const { Flame } = window.Icons;
   const { getLeaderboard } = window;
+  
+  // Get leagues that current profile is a member of
+  const userLeagues = leagues && currentProfile.leagues 
+    ? Object.values(leagues).filter(league => currentProfile.leagues.includes(league.id))
+    : [];
+  
+  // Determine which leaderboard to show
+  const leaderboardData = currentLeagueId && leagueLeaderboard
+    ? getLeaderboard(null, currentLeagueId, leagueLeaderboard)
+    : getLeaderboard(users);
 
   return (
     <div className="bg-white rounded-lg shadow-md p-6">
-      <h2 className="text-xl font-bold mb-4 text-gray-800">FamiLes ledertavle</h2>
+      <div className="mb-4">
+        <h2 className="text-xl font-bold mb-4 text-gray-800">
+          {currentLeagueId ? 'Liga ledertavle' : 'FamiLes ledertavle'}
+        </h2>
+        
+        {/* League Selector */}
+        {userLeagues.length > 0 && (
+          <div className="mb-4">
+            <label className="block text-sm font-medium text-gray-700 mb-2">
+              Velg liga
+            </label>
+            <select
+              value={currentLeagueId || ''}
+              onChange={(e) => setCurrentLeagueId(e.target.value || null)}
+              className="w-full px-4 py-2 border border-gray-300 rounded-lg focus:outline-none focus:ring-2 focus:ring-indigo-500"
+            >
+              <option value="">Global ledertavle</option>
```

Comment:
There shouldn't be a global leaderboard. If a user hasn't joined any league yet, the leaderboard should be empty with a message about needing to join a league.